R/table.R

```diff
@@ -310,21 +310,25 @@ kable_latex = function(
   x = escape_latex_table(x, escape, booktabs)
   if (!is.character(toprule)) toprule = NULL
   if (!is.character(bottomrule)) bottomrule = NULL
-  tabular = if (longtable) 'longtable' else 'tabular'
+  tabular = if (longtable) 'longtable' else if (tabularx) 'tabularx' else if (xltabular) 'xltabular' else 'tabular'
+  if (table.length == '') table.length = '\\linewidth'
 
   paste(c(
-    if (!longtable) c(env1, cap, centering),
-    sprintf('\n\\begin{%s}%s', tabular, valign), align,
-    if (longtable && cap != '') c(cap, '\\\\'),
+    if (!longtable & !xltabular) c(env1, cap, centering),
+    if (tabularx | xltabular) sprintf('\n\\begin{%s}{%s}', tabular, table.length) else sprintf('\n\\begin{%s}%s', tabular, valign),
+    align,
+    if (longtable | xltabular && cap != '') c(cap, '\\\\'),
```

Comment:
I think the error you saw was because of this:

```suggestion
    if ((tabular %in% c('longtable', 'xltabular')) && cap != '') c(cap, '\\\\'),
```

Reply:
Yep, you are right. Now the table works.